src/engine/Operation.cpp

```diff
@@ -308,6 +308,11 @@ std::shared_ptr<const Result> Operation::getResult(
   const bool pinResult =
       _executionContext->_pinSubtrees || pinFinalResultButNotSubtrees;
 
+  // If pinned there's no point in computing the result lazily.
+  if (pinResult && computationMode == ComputationMode::LAZY_IF_SUPPORTED) {
+    computationMode = ComputationMode::FULLY_MATERIALIZED;
```

Comment:
isn't the second half of the `&&` completely redundant, and you can just write
`if (pinResult) mode = MATERIALIZED;`

(that function is already long enough).

Reply:
No, because there's also the variant `ONLY_IF_CACHED`, which should simply move an already cached value to the pinned cache.